### PR TITLE
Implement `IN` expr

### LINKED ISSUE #219: Implement `IN` expr
This issue is to track the effort for implementing `IN` expr. to ensure queries as the following evaluate correctly:

```SQL
SELECT a AS b FROM [{'a': 1}, {'a': 2}, {'a': 3}] WHERE a IN [1];
-- Expected << {'b': 1} >>
```

### DoD
- Changes are PR'ed, approved, and merged.

### PR DESCRIPTION
**Pull request by am357**
*Issue #, if available:*

Closes #219

*Description of changes:*
This commit adds the required changes to implement `IN` expression. The following test passes:

```SQL
SELECT a AS b FROM [{'a': 1}, {'a': 2}, {'a': 3}] WHERE a IN [1];
-- Expected << {'b': 1} >>
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-eval/src/eval.rs [hunk 1]
```diff
@@ -535,6 +535,7 @@ pub enum EvalBinOp {
     Div,
     Mod,
     Exp,
+    In,
```
Comment (alancai98) [inline partiql-eval/src/eval.rs:538]: nit: `In` is included under the "Arithmetic ops" section. Could include under a different heading

File: partiql-eval/src/eval.rs [hunk 2]
```diff
@@ -583,6 +584,15 @@ impl EvalExpr for EvalBinOpExpr {
...
             EvalBinOp::Mod => lhs % rhs,
+            // TODO apply the changes once we clarify the rules of coercion for `IN` RHS.
+            // See also:
+            // - https://github.com/partiql/partiql-docs/pull/13
+            // - https://github.com/partiql/partiql-lang-kotlin/issues/524
+            // - https://github.com/partiql/partiql-lang-kotlin/pull/621#issuecomment-1147754213
+            EvalBinOp::In => match rhs.is_bag() | rhs.is_list() {
+                true => Boolean(rhs.into_iter().contains(&lhs)),
+                false => Boolean(false),
+            },
```
Comment (alancai98) [inline partiql-eval/src/eval.rs:595]: Think we need to handle case of `MISSING` and `NULL` propagation of the RHS (e.g. `... IN MISSING` and `... in NULL`)

Comment (am357) [reply partiql-eval/src/eval.rs:595]: Correct, let me fix it.

File: partiql-eval/src/eval.rs (hunk 1 above)
Comment (am357) [reply partiql-eval/src/eval.rs:538]: Ack, will fix.

File: partiql-eval/src/eval.rs (hunk 2 above)
Comment (jpschorr) [reply partiql-eval/src/eval.rs:595]: This is my only quibble with this PR as well.

**Review by jpschorr — APPROVED** [review]

**Review by jpschorr — APPROVED** [review]

**Review by alancai98 — APPROVED** [review]

**Pull request merged**